src/Discord.Net.Rest/Entities/Guilds/RestGuild.cs

```diff
@@ -314,6 +321,13 @@ public Task DeleteEmoteAsync(GuildEmote emote, RequestOptions options = null)
             else
                 return ImmutableArray.Create<IVoiceChannel>();
         }
+        async Task<IReadOnlyCollection<ICategoryChannel>> IGuild.GetCategoriesAsync(CacheMode mode, RequestOptions options)
+        {
+            if (mode == CacheMode.AllowDownload)
+                return await GetCategoryChannelsAsync(options).ConfigureAwait(false);
+            else
+                return null;
```

Comment:
Return an empty ImmutableArray here, as in GetVoiceChannels.

Furthermore, this method should be below GetVoiceChannelAsync and an accompanying GetCategoryAsync should be added.

Reply:
Can also wrap the results in a `Task.FromResult()` so we avoid allocating the statemachine at runtime.

Reply:
That's fair.